### PR TITLE
add the form-data-file flag to the curl post command to send multipart/form-data requests

### PR DESCRIPTION
**Pull request by jnschbrt**
### Description
- cURL POST command extended by the `--form-data-file` flag
- use the cli as follows to perform a multipart/form-data file upload required by e.g. the saved search import API:
```
opensearch-cli curl post --profile local --path "/_dashboards/api/saved_objects/_import" --form-data-file "export.ndjson" -H 'osd-xsrf:true' -q "createNewCopies=true"
```

-  
### Issues Resolved
- https://github.com/opensearch-project/opensearch-cli/issues/76
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/opensearch-cli/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by wbeckler** [comment]
Wow, thank you for contributing!  @VachaShah would you be willing to take a look?

File: commands/curl_post.go [hunk 1]
```diff
@@ -68,6 +68,8 @@ func init() {
 	curlPostCmd.Flags().StringP(
 		curlDataFlagName, "d", "",
 		"Data for the REST API. If value starts with '@', the rest should be a file name to read the data from.")
+	curlPostCmd.Flags().StringP(curlFormDataFileFlagName, "F", "", "Specify multipart upload file. This causes curl to POST data using the Content- Type multipart/form-data according to RFC2388."+
```
Comment (VachaShah) [inline commands/curl_post.go:71]: Is the file type required to be json? If so, lets specify it in this command description?

**Review by VachaShah — COMMENTED** [review]
Thank you @jnschbrt for adding this feature!

Can we also add the usage in the [README](https://github.com/opensearch-project/opensearch-cli/blob/main/README.md)?

File: commands/curl_post.go (hunk 1 above)
Comment (jnschbrt) [reply commands/curl_post.go:71]: not necessarily. depending on what the opensearch / kibana rest api expects.
the form-data-file flag parses the file in bytes. what is in the file, or what format it has, does not matter. it is only checked whether the file exists. 😃

**Comment by jnschbrt** [comment]
Hi @VachaShah , 

have you had time to look at the new changes? :)

**Review by VachaShah — COMMENTED** [review]
The changes related to the feature look good to me. The untriaged workflow is already part of the main so we can remove it from the PR.

**Comment by jnschbrt** [comment]
> The changes related to the feature look good to me. The untriaged workflow is already part of the main so we can remove it from the PR.

removed

**Comment by VachaShah** [comment]
Hi @jnschbrt, can you merge the latest changes from main to your feature branch. The build failure was fixed in the latest commit. This will make the CI green and we can merge this PR.

**Comment by jnschbrt** [comment]
Hi @VachaShah 

merged the main into my branch

**Comment by VachaShah** [comment]
@jnschbrt Thank you for your patience on this PR! I see a build failure happening due to lint errors: https://github.com/opensearch-project/opensearch-cli/actions/runs/5069380734/jobs/9139305477?pr=77. Can you please take a look?

**Review by VachaShah — COMMENTED** [review]
LGTM! Thank you @jnschbrt for your patience on this PR.

**Review by VachaShah — APPROVED** [review]

**Pull request merged**